### ISSUE TITLE
Demo Script

### ISSUE THREAD

**Issue by ZEP-Arter** [ZEP-Arter/KNT#8]
SETUP SCRIPT
1. Hex Tile configuration - For the demo we need to hard code a preconfigured hex placement of the map @BBKING55 
2. Starting position determination - They will select predestined locations. We need to have it so that the turn placement goes with every placement until the 3rd time they place a marker.
3. Each Player begins with 10 gold pieces and 1 Fort - They will pick a predestined location for their forts
4. Each player is given hard coded armies - They will then place their armies in predestined locations

SEQUENCE OF PLAY SCRIPT

Basically we need Gold collection, Recruit Things,  Movement and Combat working.

We have to hard code a bunch of stuff for the script.

Reference: http://people.scs.carleton.ca/~jeanpier//304W14/Iteration%20One%20Demo%20Script%20-%20Version%203.pdf

**Comment by ZEP-Arter** [comment]
I think the main game loop should look like this
main()
{
  setup();

  while(noWinner)
  {
    collectGold();
    recruitCharacter();
    recruitThings();
    randomEvents();
    movementPhase();
    combatPhase();
    constructionPhase();
    specialPowersPhase();
    changeOrder();
  }
  displayWinner();
}

**Comment by ericadmski** [comment]
looks good to me. So we will need a phase base class.

**Comment by ZEP-Arter** [comment]
I have a rudimentary logic of how we can check what tiles can be selected for movement.
Which I think is the big part of movement.
Also check my Update Tile.cs, I've added a list of Things for each player on a Tile so we have places to store the data.

checkMovement(hexNum, moveLeft) 
{
traversed = true;
if(!faceup)
{
flipTile();
return;
}
if(moveLeft <= 0) return;
foreach surrounding hexNum h
{
if(!traversed)
{
if(rough & moveLeft >= 2)  checkMovement(h, moveLeft -2);
else if(!rough) checkMovement(h, moveLeft);
}
}
}

then at the end of the recursion, if branden can make the tiles with traversed highlighted and selectable thats movement done.

_If player selects another stack of things we run a reset making traversed false on all the tiles
*_same goes for when player switches
